network/p2p/scoring/decay_test.go

```diff
@@ -239,15 +242,53 @@ func TestDefaultDecayFunction(t *testing.T) {
 				},
 			},
 		},
+		{
+			// 6. penalty is negative and below slowerDecayPenaltyThreshold record decay should be adjusted.
+			name: "penalty is negative and below slowerDecayPenaltyThreshold record decay should be adjusted",
+			args: args{
+				record: p2p.GossipSubSpamRecord{
+					Penalty: -100,
+					Decay:   0.8,
+				},
+				lastUpdated: time.Now(),
+			},
+			want: want{
+				record: p2p.GossipSubSpamRecord{
+					Penalty: -100,
+					Decay:   0.81,
+				},
+			},
+		},
+		{
+			// 6. penalty is negative and below slowerDecayPenaltyThreshold but record.LastDecayAdjustment is too recent. In this case the decay should not be adjusted.
+			name: "penalty is negative and below slowerDecayPenaltyThreshold record decay should not be adjusted",
+			args: args{
+				record: p2p.GossipSubSpamRecord{
+					Penalty:             -100,
+					Decay:               0.9,
+					LastDecayAdjustment: time.Now().Add(10 * time.Second),
+				},
+				lastUpdated: time.Now(),
+			},
+			want: want{
+				record: p2p.GossipSubSpamRecord{
+					Penalty: -100,
+					Decay:   0.9,
+				},
+			},
+		},
 	}
 
-	decayFunc := scoring.DefaultDecayFunction()
+	flowConfig, err := config.DefaultConfig()
+	assert.NoError(t, err)
+	scoringRegistryConfig := flowConfig.NetworkConfig.GossipSubConfig.GossipSubScoringRegistryConfig
+	decayFunc := scoring.DefaultDecayFunction(scoringRegistryConfig.SlowerDecayPenaltyThreshold, scoringRegistryConfig.DecayRateDecrement, scoringRegistryConfig.DecayAdjustInterval)
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := decayFunc(tt.args.record, tt.args.lastUpdated)
 			assert.NoError(t, err)
-			assert.Less(t, math.Abs(got.Penalty-tt.want.record.Penalty), 10e-3)
-			assert.Equal(t, got.Decay, tt.want.record.Decay)
+			assert.Less(t, math.Abs(got.Penalty-tt.want.record.Penalty), 10e-2)
```

Comment:
```suggestion
        tolerance := 0.01 // 1% tolerance
        expectedPenalty := tt.want.record.Penalty
        
        // ensure expectedPenalty is not zero to avoid division by zero
        if expectedPenalty != 0 {
            normalizedDifference := math.Abs(got.Penalty - expectedPenalty) / math.Abs(expectedPenalty)
            assert.Less(t, normalizedDifference, tolerance)
        } else {
            // handles the case where expectedPenalty is zero
            assert.Less(t, math.Abs(got.Penalty), tolerance)
        }
```
I've also recently encountered unsustainable testing behavior when directly taking the absolute value of the difference between got and want values. Normalizing the difference suppresses their scale and yields a more sustainable behavior.